app/common/auth/__init__.py

```diff
@@ -20,15 +20,13 @@
 @auth_blueprint.route("/request-a-link-to-sign-in", methods=["GET", "POST"])
 @auto_commit_after_request
 def request_a_link_to_sign_in() -> ResponseReturnValue:
-    form = SignInForm()
+    form = SignInForm(data={"redirect_to": request.args.get("next", url_for("index"))})
```

Comment:
Does putting `next` in the tamper-proof session which is set by the handler that realises you don't have authentication leave a bit less scope for users accidentally or maliciously tampering with this? 

Looking over the method to sanitise the input I feel like it wouldn't be an issue if it was changed but good to understand

Reply:
That feels like a v sensible shout, let's do that 👍  (caveat: we don't want to bung everything into the session, but this feels fine)